docs/performance.rst

```diff
@@ -0,0 +1,108 @@
+
+Performance
+===========
+
+For-loop decorators
+-------------------
+
+In Taichi, as we all know, the outermost for-loop is parallelized.
+
+However, there're some implementation details about **how it is parallelized**.
+
+Taichi provides some API to modify these parameters which is usually invisible
+from end-users.
+This allows advanced users to manually fine tune the performance to be better.
+
+For example, specifying a suitable ``ti.block_dim`` could yield almost 3x performance boost in `examples/mpm3d.py <https://github.com/taichi-dev/taichi/blob/master/examples/mpm3d.py>`_!
+
+.. note::
+
+   For performance profiling utilities, see :ref:`profiler`.
+
+Hierarchical structure of GPU
+*****************************
+
+Taichi programs could be executed in parallel on GPU.
+The level of structure of GPU is defined hierarchically.
+
+From small to large, the computation units are:
+**invocation** < **thread** < **block** < **grid**.
```

Comment:
I think **iteration** is a better name compared to **invocation**. Please consider changing the word.

Reply:
```suggestion
**iteration** < **thread** < **block** < **grid**.
```